src/jni_core/src/jvm_runtime.rs

```diff
@@ -18,50 +18,55 @@ use std::fs;
 use std::path::Path;
 use std::sync::OnceLock;
 
+use anyhow::anyhow;
 use jni::objects::JValueOwned;
 use jni::strings::JNIString;
 use jni::{InitArgsBuilder, JNIVersion, JavaVM, NativeMethod};
-use risingwave_common::error::{ErrorCode, RwError};
 use risingwave_common::util::resource_util::memory::system_memory_available_bytes;
+use tracing::error;
 
 /// Use 10% of compute total memory by default. Compute node uses 0.7 * system memory by default.
 const DEFAULT_MEMORY_PROPORTION: f64 = 0.07;
 
-pub static JVM: JavaVmWrapper = JavaVmWrapper::new();
+pub static JVM: JavaVmWrapper = JavaVmWrapper;
+static JVM_RESULT: OnceLock<Result<JavaVM, String>> = OnceLock::new();
 
-pub struct JavaVmWrapper(OnceLock<Result<JavaVM, RwError>>);
+pub struct JavaVmWrapper;
 
 impl JavaVmWrapper {
-    const fn new() -> Self {
-        Self(OnceLock::new())
-    }
-
-    pub fn get(&self) -> Option<&Result<JavaVM, RwError>> {
-        self.0.get()
+    pub fn for_initialized_jvm<O, F: FnOnce(&'static JavaVM) -> O>(&self, f: F) -> Option<O> {
+        JVM_RESULT.get().and_then(|result| {
+            if let Ok(jvm) = result {
+                Some(f(jvm))
+            } else {
+                None
+            }
+        })
     }
 
-    pub fn get_or_init(&self) -> Result<&JavaVM, &RwError> {
-        self.0.get_or_init(Self::inner_new).as_ref()
+    pub fn get(&self) -> anyhow::Result<&'static JavaVM> {
+        match JVM_RESULT.get_or_init(|| {
+            Self::inner_new().inspect_err(|e| error!("failed to init jvm: {:?}", e))
+        }) {
+            Ok(jvm) => Ok(jvm),
+            Err(e) => Err(anyhow!("jvm not initialized properly: {:?}", e)),
+        }
```

Comment:
I think we should follow `API` semantics of the `OnceLock` since we use `OnceLock` as the internal implementation. There are 2  methods `get` and `get_or_init` in `OnceLock`, but in this PR, we use `for_initialized_jvm` and `get` corresponding instead which might cause confusion. What about following the same method naming of `OnceLock`.

Reply:
Make sense. 

I just changed the current `get` back to `get_or_init`. The original `get` will be removed since it is only used by `load_jvm_memory_stats` which is in the same mod as `JvmWrapper` and it can use the `JVM_RESULT` directly.